go/updater/trainingJobUpdater.go

```diff
@@ -48,34 +53,417 @@ type TrainingJobUpdater struct {
 	eventCh chan *trainingJobEvent
 }
 
-func initUpdater(job *padv1.TrainingJob, kubeClient kubernetes.Interface, trainingJobClient trainingJobClient.Interface) (*TrainingJobUpdater,
+// NewUpdater creates a new TrainingJobUpdater and start a goroutine to control current job.
+func NewUpdater(job *padv1.TrainingJob, kubeClient kubernetes.Interface, trainingJobClient trainingJobClient.Interface) (*TrainingJobUpdater,
 	error) {
-	jobber := &TrainingJobUpdater{
+	log.Infof("NewJobber namespace=%v name=%v", job.Namespace, job.Name)
+	updater := &TrainingJobUpdater{
 		job:               job,
 		kubeClient:        kubeClient,
 		trainingJobClient: trainingJobClient,
 		status:            job.Status,
 		eventCh:           make(chan *trainingJobEvent, 1000),
 	}
-	return jobber, nil
+	go updater.start()
+	return updater, nil
 }
 
-// NewUpdater return a trainingJobUpdater for controller.
-func NewUpdater(job *padv1.TrainingJob, kubeClient kubernetes.Interface, trainingJobClient trainingJobClient.Interface) (*TrainingJobUpdater,
-	error) {
-	log.Infof("NewJobber namespace=%v name=%v", job.Namespace, job.Name)
-	jobber, err := initUpdater(job, kubeClient, trainingJobClient)
+// Notify is used to receive event from controller. While controller receive a informer,
+// it will notify updater to process the event. It send event to updater's eventCh.
+func (updater *TrainingJobUpdater) notify(te *trainingJobEvent) {
+	select {
+	case updater.eventCh <- te:
+		lene, cape := len(updater.eventCh), cap(updater.eventCh)
+		if lene > int(float64(cape)*0.8) {
```

Comment:
Maybe put 0.8 as a constant in this file.

Reply:
Done